src/shotgunEventDaemon.py

```diff
@@ -507,10 +516,76 @@ def _mainLoop(self):
         while self._continue:
             # Process events
             events = self._getNewEvents()
+            skipPlugins = []
+
             for event in events:
                 for collection in self._pluginCollections:
-                    collection.process(event)
-                self._saveEventIdData()
+                    for plugin in collection:
+                        pluginName = plugin.getName()
+
+                        if pluginName in skipPlugins:
+                            self.log.debug(
+                                "Not processing event #%d. Plugin %s is waiting for a gap to fill.",
+                                event["id"],
+                                pluginName,
+                            )
+                            continue
+
+                        if not plugin.isActive():
+                            self.log.debug(
+                                "Not processing event #%d. Plugin %s is inactive.",
+                                event["id"],
+                                pluginName,
+                            )
+                            continue
+
+                        nextEventId = plugin.getNextUnprocessedEventId()
+
+                        if nextEventId is None:
+                            msg = (
+                                "Plugin %s has no last event marker. Sending event #%d."
+                            )
+                            self.log.debug(msg, pluginName, event["id"])
+                            plugin.process(event)
+                        elif event["id"] < nextEventId:
+                            msg = "Last processed event for plugin %s is #%d, got #%d. Skipping."
```

Comment:
would it be helpful to explain in the log why this event is being skipped?

Reply:
This feels like it can be a bunch of things that would be too complex to list in a message. Could just documenting it in comments do the trick?

Reply:
I've added this comment:

```                            # - Another plugin might have been deactivated by a bug and has now been reactivated so we're
                            #   now looping through events that were already processed by this plugin but not by the
                            #   reactivated one.
```